wallet/api/client_check_transaction.go

```diff
@@ -131,7 +131,12 @@ func (h *ClientCheckTransaction) Handle(ctx context.Context, rawParams jsonrpc.P
 	h.interactor.Log(ctx, traceID, SuccessLog, "The transaction passes the anti-spam rules.")
 
 	// Sign the payload.
-	rawInputData := wcommands.ToInputData(request, stats.LastBlockHeight)
+	rawInputData := wcommands.ToInputData(request, stats.LastBlockHeight, uint64(ttl))
+	// always set TTL if it wasn't previously set, and override the specified value if it's > the max set by network params
+	// alternatively leave zero and set TxV3?
```

Comment:
>// alternatively leave zero and set TxV3?

Can be removed. Also, we are missing the tests that assert that behavior. The tests will document this behavior.